### PR TITLE
Fix ApidManager edge cases and throttle event

### PR DESCRIPTION
**Pull request by thomas-bc**
| | |
|:---|:---|
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |
|**_Generative AI was used in this contribution (y/n)_**|  |

---
## Change Description

Four in one:

1. Fixes a bug line 26:

```diff
- if (.... && receivedSeqCount != SEQUENCE_COUNT_ERROR)
+ if (.... && expectedSequenceCount != SEQUENCE_COUNT_ERROR)
```
this is intending to guard against doing setNextSequenceCount if we already got notification (through `expectedSequenceCount = SEQUENCE_COUNT_ERROR`) that the ApidTable is full

2. reorganize the code for clarity in `getAndIncrementSeqCount` - odd control flow is an artifact of when we used to have an inline data structure instead of a Map (see https://github.com/nasa/fprime/pull/4833)

3. Remove an FW_ASSERT in favor of an event. Even though with the bugfix 1), this code should never be reachable. Still a good practice not to assert here.

4. I also added a throttle on the event, but as I'm writing this, I wonder if maybe we specifically don't want to throttle, because these events could be valuable information

### CODE REVIEW

File: Svc/Ccsds/ApidManager/ApidManager.fpp [hunk 1]
```diff
@@ -17,7 +17,9 @@ module Ccsds {
         @ Received an unregistered APID
         event ApidTableFull(invalidApidValue: U16) \
             severity warning high \
-            format "APID Table is full, cannot generate or check sequence counts for APID: {}"
+            format "APID Table is full, cannot generate or check sequence counts for APID: {}" \
+            throttle 2 \
+            every {seconds=5}
```
Comment (thomas-bc) [inline Svc/Ccsds/ApidManager/ApidManager.fpp]: Had this at first, but I'm not convinced anymore that throttling this event is actually a good thing. Looking for feedback

File: Svc/Ccsds/ApidManager/ApidManager.cpp [hunk 2]
```diff
@@ -48,17 +48,22 @@ U16 ApidManager ::getAndIncrementSeqCount(ComCfg::Apid::T apid) {
...
-    this->log_WARNING_HI_ApidTableFull(apid);
-    return SEQUENCE_COUNT_ERROR;
+    return seqCount;  // Return the current sequence count
 }
 
 void ApidManager::setNextSeqCount(ComCfg::Apid::T apid, U16 seqCount) {
     Fw::Success insertStatus = m_apidSequences.insert(apid, seqCount);
-    FW_ASSERT(insertStatus == Fw::Success::SUCCESS, static_cast<FwAssertArgType>(apid));
+    if (insertStatus != Fw::Success::SUCCESS) {
+        // This code should not be reachable with the if (expectedSequenceCount != SEQUENCE_COUNT_ERROR)
```
Comment (LeStarch) [inline Svc/Ccsds/ApidManager/ApidManager.cpp]: If it cannot be reached, then it should be an assert because it would be a failed software assumption.

**Review by LeStarch — CHANGES_REQUESTED** [review]

File: Svc/Ccsds/ApidManager/ApidManager.cpp (hunk 2 above)
Comment (thomas-bc) [reply Svc/Ccsds/ApidManager/ApidManager.cpp]: Ok. Note to self, check UTs for why they passed before and still pass. I suspect there should be an additional check missing.
Also, see if there's a way to simplify the complexity. Maybe helper function is not needed.

**Comment by thomas-bc** [comment]
Closing and will reopen another PR

**Comment by thomas-bc** [comment]
Superseded by https://github.com/nasa/fprime/pull/5091
